lucille-core/src/main/java/com/kmwllc/lucille/core/Document.java

```diff
@@ -570,4 +574,131 @@ static Document createFromJson(String json, UnaryOperator<String> idUpdater) thr
     return JsonDocument.fromJsonString(json, idUpdater);
   }
 
+  /* --- JSON Getters/Setters --- */
+
+  /**
+   * Gets a nested JsonNode at a path like "a.b.c.d" where the path is split on '.' and each part is treated as a level of nesting.
+   * This also works for nested values that contain a list, such as "a.b.2.c" where 'b' is an ArrayNode.
+   *
+   * @param name the nested field path to get the JsonNode from
+   * @return the JsonNode at the nested path or null if not found
+   */
+  default JsonNode getNestedJson(String name) {
+    String[] pathSegments = name.split("\\.");
+
+    if (pathSegments.length == 0 || !has(pathSegments[0])) {
+      return null;
+    }
+    JsonNode node;
+    try {
+      node = getJson(pathSegments[0]);
+    } catch (ClassCastException e) {
+      log.error("Field {} is not a JsonNode, cannot get nested value {}", pathSegments[0], name);
+      return null;
+    }
+
+    for (int i = 1; i < pathSegments.length; i++) {
+      String segment = pathSegments[i];
+
+      if (!hasFieldSegment(node, segment)) {
+        return null;
+      } else {
+        node = getFieldSegment(node, segment);
+      }
+    }
+
+    return node;
+  }
+
+  /**
+   * Sets a nested JsonNode at a path like "a.b.c.d" where the path is split on '.' and each part is treated as a level of nesting.
+   * This also works for nested values that contain a list, such as "a.b.2.c" where 'b' is an ArrayNode.
+   *
+   * @param name the nested field path to set the JsonNode at
+   * @param value the JsonNode to set at the nested path
+   */
+  default void setNestedJson(String name, JsonNode value) {
+    String[] pathSegments = name.split("\\.");
+
+    if (pathSegments.length == 0) {
+      return;
+    }
+    if (pathSegments.length == 1) {
+      setField(pathSegments[0], value);
+      return;
+    }
+
+    // create a default fresh node where the type is based off of the child field, either ArrayNode or ObjectNode
+    JsonNode defaultFreshNode = createNewNode(pathSegments[1]);
+    JsonNode node = has(pathSegments[0]) ? getJson(pathSegments[0]) : defaultFreshNode;
```

Comment:
I would remove l632 and do
JsonNode node = has(pathSegments[0]) ? getJson(pathSegments[0]) : createNewNode(pathSegments[1]);

so we are not creating a new node we don't need to